skimage/util/arraycrop.py

```diff
@@ -0,0 +1,179 @@
+#         else:
+#             fmt = "Unable to create correctly shaped tuple from %s"
+#             raise ValueError(fmt % (shape,))
+#
+#     else:
+#         fmt = "Unable to create correctly shaped tuple from %s"
+#         raise ValueError(fmt % (shape,))
+#
+#     # Cast if necessary
+#     if cast_to_int is True:
+#         arr = np.round(arr).astype(int)
+#
+#     # Convert list of lists to tuple of tuples
+#     return tuple(tuple(axis) for axis in arr.tolist())
+
+
+# def _validate_lengths(narray, number_elements):
+#     """
+#     Private function which does some checks and reformats pad_width and
+#     stat_length using _normalize_shape.
+#
+#     Parameters
+#     ----------
+#     narray : ndarray
+#         Input ndarray
+#     number_elements : {sequence, int}, optional
+#         The width of padding (pad_width) or the number of elements on the edge
+#         of the narray used for statistics (stat_length).
+#         ((before_1, after_1), ... (before_N, after_N)) unique number of
+#         elements for each axis.
+#         ((before, after),) yields same before and after constants for each
+#         axis.
+#         (constant,) or int is a shortcut for before = after = constant for all
+#         axes.
+#
+#     Returns
+#     -------
+#     _validate_lengths : tuple of tuples
+#         int                               => ((int, int), (int, int), ...)
+#         [[int1, int2], [int3, int4], ...] => ((int1, int2), (int3, int4), ...)
+#         ((int1, int2), (int3, int4), ...) => no change
+#         [[int1, int2], ]                  => ((int1, int2), (int1, int2), ...)
+#         ((int1, int2), )                  => ((int1, int2), (int1, int2), ...)
+#         [[int ,     ], ]                  => ((int, int), (int, int), ...)
+#         ((int ,     ), )                  => ((int, int), (int, int), ...)
+#
+#     """
+#     normshp = _normalize_shape(narray, number_elements)
+#     for i in normshp:
+#         chk = [1 if x is None else x for x in i]
+#         chk = [1 if x >= 0 else -1 for x in chk]
+#         if (chk[0] < 0) or (chk[1] < 0):
+#             fmt = "%s cannot contain negative values."
+#             raise ValueError(fmt % (number_elements,))
+#     return normshp
+
+
+def crop(ar, crop_width, copy=False, order='K'):
+    """Crop array `ar` by `crop_width` along each dimension.
+
+    Parameters
+    ----------
+    ar : array-like of rank N
+        Input array.
+    crop_width : {sequence, int}
+        Number of values to remove from the edges of each axis.
+        ``((before_1, after_1),`` ... ``(before_N, after_N))`` specifies
+        unique crop widths at the start and end of each axis.
+        ``((before, after),)`` specifies a fixed start and end crop
+        for every axis.
+        ``(n,)`` or ``n`` for integer ``n`` is a shortcut for
+        before = after = ``n`` for all axes.
+    copy : bool, optional
+        Ensure the returned array is a contiguous copy. Normally, a crop
```

Comment:
If `True`, ensure...

And yo ucan remove the last sentence, which is a bit repetitive.